lib/settlement/driip-settlement-challenge-contract.js

```diff
@@ -1,11 +1,120 @@
 'use strict';
 
+const {isRevertContractException} = require('./utils');
 const NahmiiContract = require('../contract');
+const NestedError = require('../nested-error');
 
 class DriipSettlementChallengeContract extends NahmiiContract {
     constructor(walletOrProvider) {
         super('DriipSettlementChallengeByPayment', walletOrProvider);
     }
+
+    /**
+     * Returns the proposal nonce
+     * @param {Address} address - The wallet address
```

Comment:
"Address" is not a defined type AFAIK. Would be better to use our EthereumAddress class since that handles parsing, validation and conversion.

Reply:
Have you tried using the EthereumAddress class in electron frontend btw? does it work through wepbpacking etc?

Reply:
Sorry for this lagging reply 😛 
I have just had a test on the core. The EthereumAddress works 👍 
You are right. The EthereumAddress is much safer to use since it validates the addresses as well as doing a safer comparison between addresses. I am going to use this class for the settlement related functions.